updater/internal/file/file.go

```diff
@@ -43,21 +43,41 @@ func CopyFile(logger *zap.Logger, pathIn, pathOut string, overwrite bool) error
 
 	pathOutClean := filepath.Clean(pathOut)
 	fileMode := fs.FileMode(0600)
-
 	flags := os.O_CREATE | os.O_WRONLY
 	if overwrite {
 		// If we are OK to overwrite, we will truncate the file on open
 		flags |= os.O_TRUNC
 
-		// Save old file's permissions and delete it first if it exists
-		fileOutInfo, _ := os.Stat(pathOutClean)
-		if fileOutInfo != nil {
-			fileMode = fileOutInfo.Mode()
+		// Try to save old file's permissions
+		outFileInfo, _ := os.Stat(pathOutClean)
+		if outFileInfo != nil {
+			fileMode = outFileInfo.Mode()
+		} else if useInFilePermBackup {
+			// Use the new file's permissions as a backup and don't fail on error (best chance for rollback)
+			inFileInfo, err := inFile.Stat()
+			switch {
+			case err != nil:
+				logger.Error("failed to retrieve fileinfo for input file", zap.Error(err))
+			case inFileInfo != nil:
+				fileMode = inFileInfo.Mode()
+			}
+		}
```

Comment:
I think this is right, but does it make sense to always prefer the destination file's permissions, even when `useInFilePermBackup` is true? The logic just feels off to me for some reason.

Reply:
yeah...that's a good point. If the old file exists there's really no reason we won't get it's permissions.

Reply:
yeah. it's all a little confusing for sure.  ultimately we have 3 cases where we use this copy method.
1. Backup - We want to preserve the original file's permissions on the backed up file
2. Install - We want use the existing file's permissions. If there is no existing file, we are going to use 0600 and hope for the best (this might cause issues if we have executables that we add in the future).
3. Rollback - This is a little more ambiguous. If the file currently exists, it probably doesn't matter that much if we use the existing file's permissions or the backed up file's permissions. If the file does NOT exist, then we definitely want to use the backed up file's permissions (and not the default 0600).